rasa/shared/core/trackers.py

```diff
@@ -66,27 +64,23 @@
 from rasa.shared.core.slots import AnySlot, Slot
 
 if TYPE_CHECKING:
-    from typing_extensions import TypedDict
-
     from rasa.shared.core.events import NLUPredictionData
     from rasa.shared.core.training_data.structures import Story
     from rasa.shared.core.training_data.story_writer.story_writer import StoryWriter
 
-    # precise type definition for `DialogueStateTracker.active_loop`
-    TrackerActiveLoop = TypedDict(
-        "TrackerActiveLoop",
-        {
-            LOOP_NAME: Optional[Text],
-            LOOP_INTERRUPTED: bool,
-            LOOP_REJECTED: bool,
-            TRIGGER_MESSAGE: Dict,
-        },
-        total=False,
-    )
-
     EventTypeAlias = TypeVar("EventTypeAlias", bound=Event)
 
 
+@dataclasses.dataclass
+class TrackerActiveLoop:
+    """Dataclass for `DialogueStateTracker.active_loop`."""
+
+    name: Optional[Text]
+    is_interrupted: bool
```

Comment:
a test is failing right now and it's either because the test is not set up properly (and doesn't replicate an actual possible situation) or because of legacy reasons (and in this case the code needs to be updated)

Reply:
@joejuzl do you have any idea why [the tests are failing](https://github.com/RasaHQ/rasa/runs/5602642150?check_suite_focus=true) and if my first claim is correct here?

Reply:
https://github.com/RasaHQ/rasa/pull/11003/commits/2780efd60adbbc19b9afbf68c38b9f34b6d8407b